engine/src/execution_graph/logic_controllers/BatchProcessing.h

```diff
@@ -207,13 +207,22 @@ class DataSourceSequence {
 		return std::move(ret);
 	}
 	bool wait_for_next() {
-		return is_empty_data_source || (file_index < n_files and batch_index < n_batches);
+		return is_empty_data_source || (file_index < n_files and batch_index.load() < n_batches);
```

Comment:
explain this batch_index.load shenanigans, its not very clear to me

Reply:
i made batch_index an atomic, because its being accessed by multiple threads. 
.load() is a function of an atomic to fetch its value

Reply:
makes sense